[Serverless Search] Serverless Getting Started UI Polish

joemcelroy opened the pull request:
Fixes a long list of julian's UI bugs. Tested on both stateful and serverless. See videos on visual fixes.

https://github.com/elastic/kibana/assets/49480/1a450bf6-7477-40a4-a020-a5172b56ef4c

https://github.com/elastic/kibana/assets/49480/92b40ecd-d888-4fd6-af91-045e81a1843f

Things to note:
- I had to adjust the asset path here as locally on main the images were broken (the header for example).

joemcelroy committed:
ui polish on serverless-es

joemcelroy committed:
Merge remote-tracking branch 'upstream/main' into search-es/ui-polish

joemcelroy committed:
revert asset path

joemcelroy committed:
small fixes

joemcelroy committed:
Merge remote-tracking branch 'upstream/main' into search-es/ui-polish

sphilipse commented on packages/kbn-search-api-panels/components/install_client.tsx:
```diff
@@ -73,21 +73,6 @@ export const InstallClientPanel: React.FC<InstallClientProps> = ({
       />
       <EuiSpacer />
       <Link language={language} assetBasePath={assetBasePath} />
-      <EuiSpacer />
```
question: Did we want this removed entirely?

joemcelroy replied:
We had two callouts related to "try in console" so julian asked for it to be removed.

sphilipse reviewed (commented):
One minor comment but looks good otherwise, thanks!

sphilipse reviewed (approved):
Approved! LGTM :)

joemcelroy commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into search-es/ui-polish

joemcelroy commented:
@elasticmachine merge upstream

kibanamachine committed:
Merge branch 'main' into search-es/ui-polish

walterra committed:
[ML] Transforms: Enable SLO transforms installer on CI (#165470)

## Summary

This enables the installer for SLO transforms on CI.

- For the `many_fields_transform` performance journey, it will then look
for the "Create transform" on top of the table instead of the "Create
your first transform" button. Command to test the performance journey:
`node scripts/run_performance.js --journey-path
x-pack/performance/journeys/many_fields_transform.ts --skip-warmup`
- For the functional tests that assert the empty transform list, this
adds a command to delete all transforms before running those tests.

yuliacech committed:
[Index Management] Fix encoding issue on index details page (#166882)

## Summary
Fixes https://github.com/elastic/kibana/issues/166100

This PR adds a workaround fix for the new index details page when
opening for index names with special characters, for example
`test_index%`. Because of how encoding/decoding works, we can't use the
index name as a part of the url like `/indices/${indexName}` (see for
more details). Instead we have to pass the index name in a query
parameter like `/indices/index_details?indexName=${indexName}. The
downside of this workaround is that the url semantics doesn't reflect
that the index name is mandatory for the page to work. Once
https://github.com/elastic/kibana/issues/132600 is resolved, we should
revert this workaround and use the index name as a url segment again.

Note for reviewers: The jest tests for this fix are part of
https://github.com/elastic/kibana/pull/165705

### How to test
1. Add `xpack.index_management.dev.enableIndexDetailsPage: true` to the
file `config/kibana.dev.yml` to enable the new index details page
2. Navigate to Index Management and use the "create index" button 
3. Type a name with special characters, for example `test%`
4. Click the new index name in the list and check that the details page
and all tabs work
5. Also reload the page completely and check that the page still loads
correctly

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Alison Goryachev <alisonmllr20@gmail.com>

gitstart committed:
[Search Profiler] Migrate all usages of EuiPage*_Deprecated (#163131)

MadameSheema committed:
[Security Solution] Preparing Cypress to be able to execute on a real serverless environment (#166905)

Co-authored-by: YulNaumenko <jo.naumenko@gmail.com>
Co-authored-by: Sergi Massaneda <sergi.massaneda@gmail.com>

kibanamachine committed:
skip failing test suite (#167073)

kibanamachine committed:
skip failing test suite (#167076)

maximpn committed:
[Security Solution] Refactor Cypress tests navigation utilities (#166201)

**Relates to:** https://github.com/elastic/kibana/issues/153645

## Summary

This PR makes refactoring of navigation utilities and constants to make it transparent and be able to specify correct ownership later on.

## Motivation

Engineers need to add new tests and maintain the older ones. Base building blocks of this are constants and utility functions in particular URL constants and `visit()` and `visitWithoutDataRange()` functions. It turned out a simple `visit()` helper function also passes `timerange` as a query param while only some tests need it. Having it as a base utility function (as the name suggests) it also gets used in the new code while `visitWithoutDateRange()` should be preferred. On top of that URL constants are combined in one file without clear ownership and contain a mix of legacy and new urls with some parts looking outdated as navigating to the url causes redirecting to a specific page.

Having only relevant URL constants in a common file will help to reduce confusion. As the next step constants should be split into files/folders with clear ownership.

Also having `visit()` adding no extra params (besides common for ALL the tests) will make the intention clear. Whenever a time range is needed `visitWithTimeRange()` can be used (ideally accepting a time range defined in a test). And the same stays true for any other pages, e.g. rule details page can have a utility `visitRuleDetailsPage(id: string)` containing some general waiting logic so the following actions operate on a loaded page.

## Details

As a step towards clearness and transparent ownership this PR performs refactoring of `x-pack/test/security_solution_cypress/cypress/urls/navigation.ts` and `x-pack/test/security_solution_cypress/cypress/tasks/login.ts` files. The following has been done

- all url constants in `x-pack/test/security_solution_cypress/cypress/urls/navigation.ts` were checked and updated to remove duplications, avoid redirections and grouped
- legacy urls were moved to the only one test using them to test compatibility (`x-pack/test/security_solution_cypress/cypress/e2e/urls/compatibility.cy.ts`)
- `visit()` was renamed to `visitWithTimeRange()`
- `visitWithoutDateRange()` was renamed to `visit()`
- `visit()` was refactored to accept a query string

## Next steps

It's expected teams decompose `x-pack/test/security_solution_cypress/cypress/urls/navigation.ts` into feature specific file(s)/folder(s) with assigned owners.

There is no 100% chance a generic wait for a page to be loaded helper function meet requirements for each page. It makes sense to consider adding per feature `visitFeatureAPage()` helper function containing assertions for the page to be loaded. `visitRuleDetailsPage(id: string)` was added for this purpose while waiting for page to be loaded functionality is omitted in this PR to reduce a number of changes.

Heenawter committed:
[Controls] Use new `panelMinWidth` prop in popovers (#165397)

Closes https://github.com/elastic/kibana/issues/164375

## Summary

This PR wraps up https://github.com/elastic/kibana/pull/162651 by fully
migrating to the `EuiInputPopover` for all controls - specifically, this
is made possible by the new `panelMinWidth` prop, which makes it so that
the popover can now extend past the size of the input **while
maintaining** the expected positioning.

| Before | After |
|--------|--------|
| The popover was centered underneath the control on the smallest
size:<br><br>![image](https://github.com/elastic/kibana/assets/8698078/e2814ee2-6df6-47d6-925e-9f97cb8be2a5)
| The popover is left-aligned with the start of the input and expands to
the
right:<br><br>![image](https://github.com/elastic/kibana/assets/8698078/7c698ef0-1534-43b6-ac95-9ae95f1c7613)
|
| The range slider popover could not extend past the control width,
regardless of how small that
was:<br><br>![image](https://github.com/elastic/kibana/assets/8698078/12e33967-b616-4f0a-9ded-4374d65a51b2)
| The range slider popover now also has a minimum width, which makes it
more useable on the smallest
size:<br><br>![image](https://github.com/elastic/kibana/assets/8698078/2fb844db-8f5d-44d8-a6dc-c9cb95d5a4ea)
|

### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] Any UI touched in this PR is usable by keyboard only (learn more
about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [x] Any UI touched in this PR does not create any new axe failures
(run axe in browser:
[FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/),
[Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [x] This renders correctly on smaller devices using a responsive
layout. (You can test this [in your
browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [x] This was checked for [cross-browser
compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

TattdCodeMonkey commented on packages/kbn-search-api-panels/index.tsx:
```diff
@@ -37,50 +37,57 @@ export const WelcomeBanner: React.FC<WelcomeBannerProps> = ({
-                      defaultMessage: 'Hi!',
-                    })}
-              </h2>
+            <EuiTitle className="serverlessSearchHeaderTitle" size="l">
```
I know you didn't add this, but checking I don't think this `serverlessSearchHeaderTitle` css class exists anywhere anymore. We should be able to remove this.

kibanamachine committed:
skip failing test suite (#167081)

Omolola-Akinleye committed:
[Cloud Posture]remove cnvm from benchmarks (#162337)

## Summary

Remove CNVM  package policies from  the benchmarks rules page 
[Quick Win]()
<img width="1380" alt="Screen Shot 2023-07-20 at 11 17 04 AM"
src="https://github.com/elastic/kibana/assets/17135495/e8d2d161-3c77-4fdb-ae0a-dcf7512e5a10">

kibanamachine committed:
skip failing test suite (#167090)

kibanamachine committed:
skip failing test suite (#167094)

kibanamachine committed:
[api-docs] 2023-09-23 Daily api_docs build (#167102)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/469

kibanamachine committed:
[api-docs] 2023-09-24 Daily api_docs build (#167103)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/470

kibanamachine committed:
[api-docs] 2023-09-25 Daily api_docs build (#167109)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/471

gsoldevila committed:
Fix flakiness on custom time range saved searches (#165454)

A bunch of tests on dashboards are customising some of the panels
settings and providing custom time ranges:

<img width="409" alt="image"
src="https://github.com/elastic/kibana/assets/25349407/c869c1a3-f7db-4ccd-ad00-c5403f2b4201">

Currently, the logic is not waiting for the quick toggle animation to
complete, before proceeding to select a time range.
This can cause a flaky behavior if the logic tries to customize the
range before the button is actually available, as seen on [this failed
test](https://s3.amazonaws.com/buildkiteartifacts.com/e0f3970e-3a75-4621-919f-e6c773e2bb12/0fda5127-f57f-42fb-8e5a-146b3d535916/018a4c44-5497-48b6-8521-a8a79a2f63b4/018a4c46-0e7a-4b69-9a3d-9c54c27165b0/target/test_failures/018a4c46-0e7a-4b69-9a3d-9c54c27165b0_4fcbc47e71644919129e320eea8bb3bc.html?response-content-type=text%2Fhtml&X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=ASIAQPCP3C7LZWZ5UB5F%2F20230901%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20230901T094837Z&X-Amz-Expires=600&X-Amz-Security-Token=IQoJb3JpZ2luX2VjEB0aCXVzLWVhc3QtMSJGMEQCIGCyKcVLGPUawZubNzZdt5oZNb5v0saiIuPqXwI7rmwlAiAsOj%2Fiep94v%2BYZJtLY3Gw0m%2FmK5mJw2IcIBdNKFXgK%2BCr6Awjm%2F%2F%2F%2F%2F%2F%2F%2F%2F%2F8BEAAaDDAzMjM3OTcwNTMwMyIMXOd1Hm6ks%2FNE37V0Ks4DgMUso7syv87hnPcC%2BB1soxvFFnj4JnNZc6ZgkLUe93z99iPFBUsqH%2BRbUTfSbjVOEJYBKGYuvp32xvSWsYNVPXKmcej18LC0yNi%2BBzoG2X%2Bj80g%2BbGMm6YfTncjPhOE0CHHqOWXts9nQ8WpDy8XOl0zfMtuiPjzOXHo9lvw2mgYDZIJIMV72FYB9JGg8FPbLQtD3rysLGNE0VDKgl5LCnYwhY1pwRCRHnVW41QfV0pwK%2FbjNf9HjdK31LQvMY%2FGPuB3M6O2CUZLsvLGfWBeGYHtkqb0hrL9ijO1Uo28ZSS1FytPftEdF0e1kAC9C5zD56HtYm55aktOWtaaC0XPWLdWWGUq%2FKQzhxSCiXK6ovATU3zI3yPNoZs92YBYmIPMOpEI40dCCpksjPwAMCiQd%2F9gMNKP5Qp5CbYd2Khy%2FeXaT8J7HOZCueN63O0j%2FtX1tbwfznhbr74lAcRQjueRYmwboZaGSDZUQ33lSSmyZk1V9WF9eJyt88oHvIx0q9bIjvOlW05DiNKfEFWYwfBywdGuvRU6eGMs1QcDNu33Lb%2BhymudM2JZmQKIjZOcb2l3Fzctp614owH4JcRlmF4%2BIa4xHeBdRlTMysS8bTIsgMK7axacGOqYBzIpC1wgZWJ1kZ0agLWCNaMIdUl%2B4xrr7w%2Fz0843WWMhRrvbJhDTHqk5UclF%2FSROAMe0FH2XEXiQ65ILyUPlrUMels5tfQ3Pp%2FJWPi9NsQJUQ1n9uLN%2BFPDOoMo8Uxg4%2FkG2O7yTkrIdArfA6pWN9I21gFMW%2BFZy9BMYltt5T65ZKOyYAIFGpLhgfBySIBCUMgwR1kusfDhf1%2FRTvtDKD2sJKN5a0IA%3D%3D&X-Amz-SignedHeaders=host&X-Amz-Signature=35fabe908aa7514e4a92de0ed12973af85ccfb439984fc3bdd7ef3bb8fe3419b).
(part of this [failed CI
build](https://buildkite.com/elastic/kibana-pull-request/builds/155285#018a4c46-0e7a-4b69-9a3d-9c54c27165b0))

The goal of this PR is to add a small waiting period, to make sure the
toggle animation has completed, and that the time range controls are
visible and clickable.

I used the opportunity to cleanup some "await delay millis" calls,
reusing existing logic instead.

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

stratoula committed:
[ES|QL] Fixes words wrapping (#167025)

## Summary

Fixes https://github.com/elastic/kibana/issues/166648

<img width="2083" alt="image"
src="https://github.com/elastic/kibana/assets/17003240/0374c577-dc52-4453-b220-a6dfc12d79e3">

CohenIdo committed:
[Cloud Security] Fix wrong fleet url in cloudFormation (#166130)

solves:
- https://github.com/elastic/security-team/issues/7482

choose the correct Fleet URL in case there is more than one fleet
server.

---------

Co-authored-by: Maxim Kholod <maxim.kholod@elastic.co>

maxcold committed:
[Cloud Security] fix ingest pipeline for benchmark scores index (#166966)

## Summary

during the [ILM fix](https://github.com/elastic/kibana/pull/165317) for
serverless the default pipeline for the scores index was also changed by
mistake. Reverting this change


### Checklist

- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

joemcelroy committed:
remove unused class for welcome banner title
